src/dns_parser.rs

```diff
@@ -158,7 +158,7 @@ impl PartialEq for DnsRecord {
     }
 }
 
-pub trait DnsRecordExt: fmt::Debug {
+pub trait DnsRecordExt: fmt::Debug + Send {
```

Comment:
The compiler automatically implements `Send` on appropriate types, but as we have only usages of this now, it's a little bit more concise to explicitly state it on the trait rather than on the Box (and not in the answer functions), theoretically at least, should be the same thing but I think stating it like this rather than how it was is a bit better, what do you think?

Reply:
I think it is odd to have `DnsRecordExt` trait to conform `Send`, because this trait is really just about supporting different types of DNS records, nothing related to dispatching mechanisms (e.g. Send or Sync, etc).

I can understand that moving `Send` to the trait would simplify the parameter signature in functions, but it doesn't feel right for me.